testsuite/test_interface.py

```diff
@@ -35,11 +43,30 @@ def test_submission(client):
     submission = Submission.objects.all()[0]
 
     assert submission.state == submission.STATE_NEW
-    assert submission.assignment.code == 'pc-a0'
+    assert submission.assignment.code == 'a0'
     assert submission.archive_size > 0
     assert submission.vmck_job_id > 0
 
+    count = 0
     while submission.state != submission.STATE_DONE:
-        time.sleep(2)
+        submission = Submission.objects.all()[0]
```

Comment:
You can call [refresh_from_db](https://docs.djangoproject.com/en/3.0/ref/models/instances/#django.db.models.Model.refresh_from_db) to update the model instance.